zutil.h

```diff
@@ -261,6 +261,31 @@ void Z_INTERNAL   zng_cfree(void *opaque, void *ptr);
 #  endif
 #endif
 
+/* GCC < 7 doesn't optimize memcmp with unaligned instructions if the architecture
+ * supports it. This can result in negative performance impact of around 36%. */
+#if defined(UNALIGNED_OK) && (defined(__GNUC__) && !defined(__clang__) && __GNUC__ < 7)
```

Comment:
@Dead2:

- Should we only do this for memcmp? Or for both memcmp and memcpy? It really only affects memcmp.
- What do you think if we had a `FORCE_UNALIGNED` build option that would enable this in CMake/Configure? Is that preferrable?

Reply:
If we can avoid doing it for memcpy without a significant performance impact, then there is really nothing to debate.

`FORCE_UNALIGNED` might be a good idea, or `ALLOW_UNALIGNED` perhaps, but I think that change should be in a separate PR in any case, so this one stays a simple cleanup PR and not something requiring a lot of discussion.